client/termdb/TermdbVocab.js

```diff
@@ -743,64 +745,64 @@ export class TermdbVocab extends Vocab {
 					genome: this.vocab.genome,
 					dslabel: this.vocab.dslabel,
 					// one request per term
-					terms: [copy],
+					terms: copies,
 					filter,
 					embedder: window.location.hostname
 				}
 			}
 			if (opts.filter0) init.body.filter0 = opts.filter0 // avoid adding "undefined" value
 			if (opts.isHierCluster) init.body.isHierCluster = true // special arg from matrix, just pass along
-
-			if (tw.term.id && currentGeneNames?.length) {
+			if (tws.find(tw => tw.term.id && currentGeneNames?.length)) {
 				/* term.id is present meaning term is dictionary term (FIXME if this is unreliable)
 				and there are gene terms, add this to limit to mutated cases
 				*/
 				init.body.currentGeneNames = currentGeneNames
 			}
-
 			promises.push(
 				dofetch3('termdb', init).then(data => {
 					if (data.error) throw data.error
 					if (!data.refs.bySampleId) data.refs.bySampleId = {}
-					const idn = 'id' in tw.term ? tw.term.id : tw.term.name
-
-					for (const sampleId in data.samples) {
-						samplesToShow.add(sampleId)
-						const sample = data.samples[sampleId]
-						if (!(sampleId in samples)) {
-							// normalize the expected data shape
-							if (!data.refs.bySampleId[sampleId]) data.refs.bySampleId[sampleId] = {}
-							if (typeof data.refs.bySampleId[sampleId] == 'string')
-								data.refs.bySampleId[sampleId] = { label: data.refs.bySampleId[sampleId] }
-
-							const _ref_ = data.refs.bySampleId[sampleId]
-							// assign default sample ref values here
-							// TODO: may assign an empty string value if not allowed to display sample IDs or names ???
-							if (!_ref_.label) _ref_.label = sampleId
-							const s = {
-								sample: sampleId,
-								// must reserve _ref -> should not be used as a term.id or tw.$id
-								_ref_
+					for (const tw of tws) {
+						const idn = 'id' in tw.term ? tw.term.id : tw.term.name
+
+						for (const sampleId in data.samples) {
+							samplesToShow.add(sampleId)
+							const sample = data.samples[sampleId]
+							if (!(sampleId in samples)) {
+								// normalize the expected data shape
+								if (!data.refs.bySampleId[sampleId]) data.refs.bySampleId[sampleId] = {}
+								if (typeof data.refs.bySampleId[sampleId] == 'string')
+									data.refs.bySampleId[sampleId] = { label: data.refs.bySampleId[sampleId] }
+
+								const _ref_ = data.refs.bySampleId[sampleId]
+								// assign default sample ref values here
+								// TODO: may assign an empty string value if not allowed to display sample IDs or names ???
+								if (!_ref_.label) _ref_.label = sampleId
+								const s = {
+									sample: sampleId,
+									// must reserve _ref -> should not be used as a term.id or tw.$id
+									_ref_
+								}
+								samples[sampleId] = s
 							}
-							samples[sampleId] = s
+							const row = samples[sampleId]
+							if (idn in sample) row[tw.$id] = sample[idn]
```

Comment:
@airenzp The `idn` may not be unique within the tws/copies, for example if the same term like `agedx` is submitted twice, once as discrete and another as continuous. The `tws/copies` must not submit the same term more than once in the same request, in order to avoid this issue.

Reply:
@airenzp In older implementation of `getAnnotatedSampleData()`, the code detected if the term id or name already exists in the `copies`, and if so, move that term to the end of the tws array so that it will be processed in a subsequent request batch. This way, each request will only have one copy of the same term.